packages/backend-common/src/middleware/statusCheckHandler.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NextFunction, Request, Response, RequestHandler } from 'express';
+
+export interface StatusCheckHandlerOptions {
+  /**
+   * Optional status function to check the status condition.
+   */
+  status?: () => Promise<boolean>;
```

Comment:
True. That might simplify and avoid the need for the `503` because any failure would just be a `5xx`. I also considered having it return maybe a message that could be included in the check. It could work like:

```ts
try {
  // status?: () => Promise<string>;
  const statusMessage: string = await checkStatus();
  response.status(200).header('').send(statusMessage);      
} catch (err) {
  next(err);
}
```

Thoughts?